src/main/java/de/thomas_oster/liblasercut/drivers/Ruida.java

```diff
@@ -0,0 +1,1406 @@
+import java.net.DatagramPacket;
+import java.net.DatagramSocket;
+import java.net.Socket;
+import java.net.SocketTimeoutException;
+import java.net.UnknownHostException;
+import java.net.URI;
+import java.net.URISyntaxException;
+
+/* for class Serial; */
+import java.util.concurrent.TimeUnit;
+import purejavacomm.CommPort;
+import purejavacomm.CommPortIdentifier;
+import purejavacomm.NoSuchPortException;
+import purejavacomm.PortInUseException;
+import purejavacomm.PureJavaIllegalStateException;
+import purejavacomm.SerialPort;
+import purejavacomm.UnsupportedCommOperationException;
+
+public class Ruida extends LaserCutter
+{
+  private static final int MINFOCUS = -500; //Minimal focus value (not mm)
+  private static final int MAXFOCUS = 500; //Maximal focus value (not mm)
+  private static final int MAXPOWER = 80;
+  private static final double FOCUSWIDTH = 0.0252; //How much mm/unit the focus values are
+  protected static final String SETTING_HOST = "IP/Hostname";
+  protected static final String SETTING_COMPORT = "USB device";
+  protected static final String SETTING_SERIAL_TIMEOUT = "Milliseconds to wait for response";
+  protected static final String SETTING_FILE_EXPORT_PATH = "Path to save exported code";
+  protected static final String SETTING_UPLOAD_METHOD = "Upload method";
+  protected static final String UPLOAD_METHOD_FILE = "File";
+  protected static final String UPLOAD_METHOD_IP = "IP";
+  protected static final String UPLOAD_METHOD_SERIAL = "USB";
+
+  protected static final String SETTING_MAX_VECTOR_CUT_SPEED = "Max vector cutting speed (mm/s)";
+  protected static final String SETTING_MAX_VECTOR_MOVE_SPEED = "Max vector move speed (mm/s)";
+  protected static final String SETTING_MIN_POWER = "Min laser power (%)";
+  protected static final String SETTING_MAX_POWER = "Max laser power (%)";
+  protected static final String SETTING_BED_WIDTH = "Bed width (mm)";
+  protected static final String SETTING_BED_HEIGHT = "Bed height (mm)";
+  protected static final String SETTING_USE_BIDIRECTIONAL_RASTERING = "Use bidirectional rastering";
+  protected static final Locale FORMAT_LOCALE = Locale.US;
+  // config values
+  private static final long[] JogAcceleration = {200000,50000,600000};
+  private static final long[] JogMaxVelocity = {16,16,2048};
+  private static final long[] EngraveAcceleration = {200000,50000,600000};
+  private static final long[] EngraveMaxVelocity = {800,800,2048};
+  private static final long[] VectorAcceleration = {100000,25000,20000};
+  private static final long[] VectorMaxVelocity = {1000,1000,1000};
+//  private static final byte FlipLaserPWMPower = 1;
+//  private static final byte FlipLaserOutput = 0;
+
+//  private static final byte HomeDirection = 1;
+//  private static final byte[] FlipHomeDirection = {1,0,0};
+//  private static final byte[] LimitContCondition = {0,0,0,0};
+//  private static final long[] MaxSteps = {250,500,500};
+//  private static final long[] TableSize = {20000,12000,30000};
```

Comment:
please check if commented-out values are still relevant

Reply:
done. they aren't. Not sure where these came from 🤔

Reply:
then maybe remove them? :-D

Reply:
done